### ISSUE TITLE
Lane assist: selected arrow not always colored correctly

### ISSUE THREAD

**Issue by Penegal** [Project-OSRM/osrm-frontend-v2#279]
Hello, there.

![image](https://user-images.githubusercontent.com/13777824/41836662-b5088112-785b-11e8-9ff9-1cac56732b84.png)

As you can see, the last instruction before arrival includes lane assist, but does not correctly highlight the lane to be used, unlike the instruction above it. I first opened this ticket on Project-OSRM/osrm-backend#5121 but was told the problem lied in osrm-frontend.

Regards.

**Comment by yuryleb** [comment]
I don't think this is really frontend issue - the 'Keep left' maneuver from backend is `slight left` but `turn:lanes` on [that link](https://www.openstreetmap.org/way/125885850#map=18/48.69009/6.11694) are `through|slight_right`. How frontend should suppose that `slight left` means `through` here? :wink:

Maybe it's just necessary to update `turn:lanes` right on OSM to `slight_left|slight_right`? @Penegal, what are real lane directions on that road?

**Comment by danpat** [comment]
The OSRM API response returns `valid: true` for the lanes that should be preferred.  Here's what we're returning for the "Keep left" maneuver:

<img width="717" alt="screen shot 2018-06-26 at 7 05 43 am" src="https://user-images.githubusercontent.com/1892250/41917542-83df910c-790f-11e8-8124-58ff79bafb06.png">

The left-most lane has the `valid: true` indication.  The intent of this flag is to describe which lanes can be used for the maneuver.

However, I *think* LRM is having a problem with the rendering.  Compare the right lane in the "Take exit A 31" step to the left lane in the "Keep left at the fork" step:

<img width="810" alt="screen shot 2018-06-26 at 7 08 06 am" src="https://user-images.githubusercontent.com/1892250/41917827-fb44375c-790f-11e8-8fa1-3548c6e824b6.png">

The bug lies here I think:

https://github.com/Project-OSRM/osrm-frontend/blob/gh-pages/src/itinerary_builder.js#L93-L95

where we're doing as @yuryleb suggest - greying out the icon if "through" doesn't match "slight left", even though OSRM has determined already that the lane is `valid: true`.  I don't think this is actually the right thing to do, we should trust the `valid: true` indication in the API response.

**Comment by yuryleb** [comment]
Thanks @danpat for explanation but if I remove my hacks for opacity and maneuver icon, the `invalid` CSS does gray out lane icons:
![french lanes](https://user-images.githubusercontent.com/4529411/41921854-5693df2a-7953-11e8-8db9-56221483dc58.png)

`invalid` style fragment:
```
.invalid.lanes.leaflet-routing-icon {
	filter: invert(50%);
```
Maybe we should use `opacity` instead of `filter` there?

**Comment by yuryleb** [comment]
Yes, we can :blush:
![french lanes](https://user-images.githubusercontent.com/4529411/41922719-442764c2-7955-11e8-84e7-93e1a4f73ea8.png)

**Comment by danpat** [comment]
The intent of the `invert(50%)` is to convert the original icon (darkblue) in the SVG into a grey version of itself.  The opacity set [here](https://github.com/Project-OSRM/osrm-frontend/blob/gh-pages/src/itinerary_builder.js#L93-L95) then fades that.

If we remove https://github.com/Project-OSRM/osrm-frontend/blob/gh-pages/src/itinerary_builder.js#L93-L95 completely, and change the `.invalid` CSS rule to:

```css
.invalid.lanes.leaflet-routing-icon {
	filter: grayscale(100%) opacity(50%);
}
```

Then I think it might achieve what's desired here (valid lanes are blue, invalid lanes are light grey).

I saw your comment on the `osrm-backend` ticket at https://github.com/Project-OSRM/osrm-backend/issues/5121#issuecomment-400202969 and I think you might be right.  Not only are we incorrectly fading the icons in the UI, I think there is a different, but related bug in turn lane info when we're collapsing instructions.  I think we should keep those problems separate though, they have different solutions.

In the UI case, I don't think we should be separating the idea of greying the icon vs fading the icon, which is what the code is currently doing.

**Comment by yuryleb** [comment]
Yes, we can :blush:
![french lanes 2](https://user-images.githubusercontent.com/4529411/41922790-69a4a21e-7955-11e8-86b1-63032a7e9858.png)

**Comment by yuryleb** [comment]
@danpat, it seems `filter` doesn't work in IE11 :worried: Maybe we could use simple `opacity` instead?

**Comment by danpat** [comment]
Gah, and that's the reason I avoid doing frontend browser stuff.

👍 for just fading the icons and not also trying to make them grey in IE11.

**Comment by yuryleb** [comment]
I suppose the issue could be closed after merged #280
